File: go-sort/select_sort.go
package gosort

func SelectSort(arr []int) {
	for i := 0; i < len(arr)-1; i++ {
		index := i
		for j := i + 1; j < len(arr); j++ {
			if arr[index] > arr[j] {
				index = j
			}
		}
		if index != i {
			arr[index], arr[i] = arr[i], arr[index]
		}
	}
}

File: ts-sort/selectSort.ts
export default function selectSort(arr: number[]) {
  const n = arr.length
  for (let i = 0; i < n; i++) {
    let index = i
    for (let j = i + 1; j < n; j++) {
      if (arr[index]! > arr[j]!) {
        index = j
      }
    }
    if (index !== i) {
      const temp = arr[i]!
      arr[i] = arr[index]!
      arr[index] = temp
    }
  }
}
